pip/index.py

```diff
@@ -842,18 +856,37 @@ def clean_link(self, url):
 
 class Link(object):
 
-    def __init__(self, url, comes_from=None):
+    def __init__(self, url, comes_from=None, requires_python=None):
+        """
+        Object representing a parsed link from https://pypi.python.org/simple/*
+
+        url:
+            url of the resource pointed to (href of the link)
+        comes_from:
+            <Not sure>
```

Comment:
Looking at the code that instantiates it, it appears that `comes_from` is the `HTMLPage` instance from which the link was found.